dev-tools/mage/crossbuild.go

```diff
@@ -374,3 +415,21 @@ func chownPaths(uid, gid int, path string) error {
 		return nil
 	})
 }
+
+// AssembleDarwinUniversal merges the darwin/amd64 and darwin/arm64 into a single
+// universal binary using `lipo`. It assumes the darwin/amd64 and darwin/arm64
+// were built and only performs the merge.
```

Comment:
It also looks like it assumes that `lipo` is present in your `PATH`. Does this imply that this can be only be run macos machines with an xcode toolchain installed?

If so it would be good to document this assumption and possibly add an [`LookPath`](https://pkg.go.dev/os/exec#LookPath) check that provides a descriptive error about the requirements and how to get the command.

Reply:
IIUC this is automatically invoked by the `CrossBuild` target when darmin/amd64 and darmin/arm64 are platform targets. And it gets execute inside the crossbuilder container. If that's correct that would be good to mention in godoc.